xml/dm_registers.xml

```diff
@@ -571,33 +571,48 @@
         This register contains DM control and status bits that didn't easily
         fit in \RdmDmcontrol and \RdmDmstatus. All are optional.
 
-        <field name="0" bits="31:11" access="R" reset="0" />
+        If halt groups are not implemented, then \FdmDmcsTwoGroup will always
+        be 0 when \FdmDmcsTwoGrouptype is 0.
+
+        If resume groups are not implemented, then \FdmDmcsTwoGrouptype will
+        remain 0 even after 1 is written there.
+
+        The triggers referred to in \FdmDmcsTwoExttrigger might be the same,
+        or might be completely separate sets, because it is not generally
+        useful to automatically halt some cores when other cores resume
+        or vice versa.
+
+        <field name="0" bits="31:12" access="R" reset="0" />
+        <field name="grouptype" bits="11" access="WARL" reset="0">
+            0: The remaining fields in this register configure halt groups.
+
+            1: The remaining fields in this register configure resume groups.
+        </field>
         <field name="exttrigger" bits="10:7" access="WARL" reset="0">
             This field contains the currently selected external trigger.
 
             If a non-existent trigger value is written here, the hardware will
             change it to a valid one or 0 if no external triggers exist.
         </field>
-        <field name="haltgroup" bits="6:2" access="WARL" reset="0">
-            When \FdmDmcsTwoHgselect is 0, contains the halt group of the hart
+        <field name="group" bits="6:2" access="WARL" reset="0">
+            When \FdmDmcsTwoHgselect is 0, contains the group of the hart
             specified by \Fhartsel.
 
-            When \FdmDmcsTwoHgselect is 1, contains the halt group of the external
+            When \FdmDmcsTwoHgselect is 1, contains the group of the external
             trigger selected by \FdmDmcsTwoExttrigger.
 
             Writes only have an effect if \FdmDmcsTwoHgwrite is also written 1.
 
             An implementation may tie any number of upper bits in this field to
-            0. If halt groups aren't implemented, then this entire field
-            is 0.
+            0. If groups aren't implemented, then this entire field is 0.
```

Comment:
Should we specifically state that the number of halt groups need not be equal to the number of resume groups?  For example, writing a value of 2 in the "group" field might be legal if "grouptype" is being written to 0 but not if "grouptype" is being written to 1.  Maybe the "WARL" is sufficient for the spec but in this case, it's not just this field alone that's WARL - the legal values for this field depend on other fields.